evap/staff/importers.py

```diff
@@ -86,18 +87,34 @@ def store_in_database(self, vote_start_date, vote_end_date, semester):
 
 
 class ExcelImporter(object):
+    W_NAME = 'name'
+    W_EMAIL = 'email'
+    W_DUPL = 'duplicate'
+    W_GENERAL = 'general'
+
     def __init__(self, request):
         self.associations = OrderedDict()
         self.request = request
         self.book = None
         self.skip_first_n_rows = 1  # first line contains the header
         self.errors = []
-        self.warnings = []
+
+        self.warnings = {self.W_NAME: [], self.W_EMAIL: [], self.W_GENERAL: [], self.W_DUPL: []}
+
         # this is a dictionary to not let this become O(n^2)
         self.users = {}
 
     def read_book(self, excel_file):
-        self.book = xlrd.open_workbook(file_contents=excel_file.read())
+        if excel_file is None:
+            self.errors.append(_("Please select an Excel file."))
+            return
+        try:
+            if isinstance(excel_file, UploadedFile):
+                self.book = xlrd.open_workbook(file_contents=excel_file.read())
+            else:
+                self.book = xlrd.open_workbook(excel_file)
```

Comment:
Changed it so the importer is given the file's content and doesn't handle the possble difference of whether it's given an UploadedFile or a file path.